### PR TITLE
Carla navigation sensor

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/environments/suite_carla.py [hunk 1]
```diff
@@ -689,6 +731,70 @@ def transform_to_geolocation(self, location: carla.Location):
         return np.array([loc.latitude, loc.longitude, loc.altitude])
 
 
+class NavigationSensor(SensorBase):
+    """Generating future waypoints on the route.
+
+    Note that the route is fixed (not change based on current vehicale location)
```
Comment (le-horizon) [inline alf/environments/suite_carla.py]: ->vehicle

**Review by le-horizon — DISMISSED** [review]
Took a quick look, and looks good.  Let's merge it in~

File: alf/environments/suite_carla.py [hunk 2]
```diff
@@ -638,6 +666,20 @@ def __init__(self, world: carla.World):
         self._vehicle_dict = {}
         self._vehicle_locations = {}
 
+        dao = GlobalRoutePlannerDAO(world.get_map(), sampling_resolution=1.0)
+        self._global_route_planner = GlobalRoutePlanner(dao)
+        self._global_route_planner.setup()
+
+    def trace_route(self, origin, destination):
+        """
```
Comment (hnyu) [inline alf/environments/suite_carla.py]: comment

File: alf/environments/suite_carla.py [hunk 3]
```diff
@@ -689,8 +731,80 @@ def transform_to_geolocation(self, location: carla.Location):
...
+
+    def observation_spec(self):
+        return alf.TensorSpec([len(self._future_indices), 3])
+
+    def observation_desc(self):
+        return ("Positions of the %s future locations in the route." % len(
+            self._future_indices))
+
+    def get_current_observation(self, current_frame):
+        """Get the current observation.
```
Comment (hnyu) [inline alf/environments/suite_carla.py:777]: More descriptive? Might copy `observation_desc` here. Also explain the observation spec.

File: alf/environments/suite_carla.py [hunk 4]
```diff
@@ -689,8 +731,80 @@ def transform_to_geolocation(self, location: carla.Location):
...
+def _to_numpy_loc(loc: carla.Location):
+    return np.array([loc.x, loc.y, loc.z])
+
+
 class Player(object):
-    """Player is a vehicle with some sensors."""
+    """Player is a vehicle with some sensors.
+
+    An episode terminates if the vehicle arrives at the goal or the time exceeds
+    `initial_distance / min_speed `.
```
Comment (hnyu) [inline alf/environments/suite_carla.py]: \``

File: alf/environments/suite_carla.py [hunk 5]
```diff
@@ -689,8 +731,80 @@ def transform_to_geolocation(self, location: carla.Location):
...
+
+    At each step, the reward is given based on the following components:
+    1. Arriving goal:  `success_reward`
```
Comment (hnyu) [inline alf/environments/suite_carla.py]: \``

**Review by hnyu — CHANGES_REQUESTED** [review]

**Review by hnyu — APPROVED** [review]

**Pull request merged**